hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -792,11 +798,9 @@ boolean areDiffJobAndSnapshotsActive(
     SnapshotInfo toSnapInfo = getSnapshotInfo(ozoneManager, volumeName,
         bucketName, toSnapshotName);
 
-    String jobKey = fromSnapInfo.getSnapshotId() +
-        DELIMITER + toSnapInfo.getSnapshotId();
-
-    if (snapDiffJobTable.get(jobKey).getStatus()
-        .equals(CANCELLED)) {
+    String jobKey = generateSnapDiffJobKey.apply(fromSnapInfo, toSnapInfo);
+    SnapshotDiffJob diffJob = snapDiffJobTable.get(jobKey);
+    if (diffJob == null || diffJob.getStatus() == CANCELLED) {
```

Comment:
should job status DONE, REJECTED, FAILED also return false?

Reply:
DONE, REJECTED and FAILED should not come to this path. Because one job is single thread and once it is marked DONE, REJECTED or FAILED, it won't come to this flow. You can create new job after that cleanup of REJECTED or FAILED but that would be considered a different job altogether.

Reply:
I see, thanks Hemant!